relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -1,116 +1,73 @@
 //! Functionality for calculating if a trace should be processed or dropped.
-use chrono::{DateTime, Utc};
+use chrono::Utc;
 use relay_base_schema::project::ProjectKey;
-use relay_event_schema::protocol::Event;
-use relay_sampling::evaluation::{MatchedRuleIds, SamplingMatch};
-use relay_sampling::DynamicSamplingContext;
+use relay_sampling::config::{RuleType, SamplingMode};
+use relay_sampling::evaluation::{RuleMatchingState, SamplingEvaluator, SamplingMatch};
+use relay_sampling::{DynamicSamplingContext, SamplingConfig};
 
-use crate::actors::project::ProjectState;
 use crate::envelope::{Envelope, ItemType};
 
-/// The result of a sampling operation.
-#[derive(Debug, Default, Clone, PartialEq, Eq)]
+/// Represents the specification for sampling an incoming event.
+#[derive(Default, Clone, Debug, PartialEq)]
 pub enum SamplingResult {
-    /// Keep the event.
-    ///
-    /// Relay either applied sampling rules and decided to keep the event, or was unable to parse
-    /// the rules.
+    /// The event matched a sampling condition.
+    Match(SamplingMatch),
+    /// The event did not match a sampling condition.
+    NoMatch,
+    /// The event has yet to be run a dynamic sampling decision.
     #[default]
-    Keep,
-
-    /// Drop the event, due to a list of rules with provided identifiers.
-    Drop(MatchedRuleIds),
+    Pending,
 }
 
 impl SamplingResult {
-    fn determine_from_sampling_match(sampling_match: Option<SamplingMatch>) -> Self {
```

Comment:
moved to `fn sampling_match` which is called in the constructor of `SamplingMatch`